gamemodes/terrortown/entities/weapons/weapon_ttt_wtester.lua

```diff
@@ -154,710 +216,322 @@ function SWEP:GatherRagdollSample(ent)
 
-else -- SERVER
-	local function ScanPrint(ply, cmd, args)
-		local tester = GetTester(ply)
-
-		if not IsValid(tester) then return end
-		if #args ~= 1 then return end
-
-		local i = tonumber(args[1])
-
-		if not i then return end
+		-- add an empty line if there's already data in the description area
+		if tData:GetAmountDescriptionLines() > 0 then
+			tData:AddDescriptionLine()
+		end
 
-		tester:PerformScan(i)
-	end
-	concommand.Add("ttt_wtester_scan", ScanPrint)
+		if ent:IsWeapon() or ent.CanHavePrints or ent:GetNWBool("HasPrints", false)
+		or ent:GetClass() == "prop_ragdoll" and CORPSE.GetPlayerNick(ent, false)
+		then
+			tData:AddDescriptionLine(TryT("dna_tid_possible"), COLOR_GREEN, {dna_icon})
+		else
+			tData:AddDescriptionLine(TryT("dna_tid_impossible"), COLOR_RED, {dna_icon})
+		end
+	end)
 
-	local function RemoveSample(ply, cmd, args)
-		local tester = GetTester(ply)
+	local function ScannerFeedback()
+		if not LocalPlayer():HasWeapon("weapon_ttt_wtester") then return end
 
-		if #args ~= 1 then return end
+		local scanner = LocalPlayer():GetWeapon("weapon_ttt_wtester")
 
-		local idx = tonumber(args[1])
+		local successful = net.ReadBool()
+		local oldFound = net.ReadBool()
 
-		if not idx or not IsValid(tester) then return end
+		if successful or oldFound then
+			scanner.ActiveSample = net.ReadUInt(8)
+			scanner.ItemSamples[scanner.ActiveSample] = net.ReadEntity()
+			scanner.NewSample = scanner.ActiveSample
 
-		tester:RemoveItemSample(idx)
+			scanner:RadarScan()
+		end
+		scanner.ScanTime = CurTime()
+
+		if successful then
+			scanner:EmitSound(beep_success)
+			scanner.ScanSuccess = 1
+		elseif oldFound then
+			scanner:EmitSound(beep_match)
+			scanner.ScanSuccess = 2
+		else
+			scanner:EmitSound(beep_miss)
+			scanner.ScanSuccess = 0
+		end
 	end
-	concommand.Add("ttt_wtester_remove", RemoveSample)
-end
-
-function SWEP:OnRemove()
-	if SERVER then return end
+	net.Receive("TTT2ScannerFeedback", ScannerFeedback)
 
-	self:ClosePrintsPanel()
-end
-
-function SWEP:OnDrop()
+	local function ScannerUpdate()
+		if not LocalPlayer():HasWeapon("weapon_ttt_wtester") then return end
 
-end
+		local scanner = LocalPlayer():GetWeapon("weapon_ttt_wtester")
 
-function SWEP:PreDrop()
-	local owner = self:GetOwner()
-
-	if not IsValid(owner) then return end
-
-	owner.scanner_weapon = nil
-end
-
-function SWEP:Reload()
-	return false
-end
-
-function SWEP:Deploy()
-	local owner = self:GetOwner()
-
-	if SERVER and IsValid(owner) then
-		owner:DrawViewModel(false)
-		owner.scanner_weapon = self
+		local idx  = net.ReadUInt(8)
+		scanner.ItemSamples[idx] = net.ReadEntity()
 	end
-
-	return true
+	net.Receive("TTT2ScannerUpdate", ScannerUpdate)
 end
 
-if CLIENT then
-	function SWEP:DrawWorldModel()
-		if IsValid(self:GetOwner()) then return end
+hook.Add("Tick", "TTT2DNAScannerThink", function()
+	plys = player.GetAll()
```

Comment:
I don't checked, but it is right not localizing it?

Reply:
how to localize it? outside of the hook? could be problematic imo
or do you mean just doing `local player_GetAll = player.GetAll`?

Reply:
I mean `local plys = player.GetAll()`

Reply:
Ah, yes, I see now. Why is the `local` missing here?